Libraries/Image/RCTImageLoader.m

```diff
@@ -480,6 +470,18 @@ - (RCTImageLoaderCancellationBlock)loadImageWithURLRequest:(NSURLRequest *)image
   __block void(^cancelLoad)(void) = nil;
   __weak RCTImageLoader *weakSelf = self;
 
+  // Load handlers are executed before the generic caching logic so they are responsible
+  // for caching the images if necessary.
+  id<RCTImageURLLoader> loadHandler = [self imageURLLoaderForURL:imageURLRequest.URL];
```

Comment:
Actually isn't this approach less memory intensive but potentially slower? We are loading images one by one on the main thread so there should only ever be one in memory at a given time. The url queue is useful for loading images asynchronously since there may be many image decoding happening at the same time so huge peaks of memory usage. The only thing that could happen is that many images are loaded on the url queue at the same time while images are loading on the main thread but since the main thread is loading only one image it should be an issue.